Dalamud/Interface/Internal/ReShadeHandling/ReShadeUnwrapper.cs

```diff
@@ -0,0 +1,205 @@
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+using TerraFX.Interop.Windows;
+
+using static TerraFX.Interop.Windows.Windows;
+
+namespace Dalamud.Interface.Internal.ReShadeHandling;
+
+/// <summary>
+/// Peels ReShade off stuff.
+/// </summary>
+[SuppressMessage(
+    "StyleCop.CSharp.LayoutRules",
+    "SA1519:Braces should not be omitted from multi-line child statement",
+    Justification = "Multiple fixed blocks")]
+internal static unsafe class ReShadeUnwrapper
```

Comment:
I guess this seems to work, I'm curious how stable it is. Will this break if the inheritance hierarchy of those COM objects changes?

Reply:
The interface definitions are part of Windows API, so they will not change. They are guaranteed to be based on IUnknown, so I don't see this breaking anytime soon. The only possible breakage is when ReShade wrapper object places pointer to the original COM object past 3rd member field or creates another indirection to it.